### PR TITLE
Liveliness QoS API [5106]

### PR DESCRIPTION
**Pull request by raquelalvarezbanos**
This PR adds methods/structs at the pub/sub level related to liveliness QoS

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1495/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1461/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1472/)

File: include/fastrtps/publisher/PublisherListener.h [hunk 1]
```diff
@@ -37,12 +38,32 @@ class RTPS_DllAPI PublisherListener
...
 	PublisherListener(){};
 	virtual ~PublisherListener(){};
+
 	/**
 	 * This method is called when the Publisher is matched (or unmatched) against an endpoint.
 	 * @param pub Pointer to the associated Publisher
 	 * @param info Information regarding the matched subscriber
 	 */
-	virtual void onPublicationMatched(Publisher* pub, rtps::MatchingInfo& info){(void)pub; (void)info;};
+    virtual void onPublicationMatched(
```
Comment (richiware) [inline include/fastrtps/publisher/PublisherListener.h]: Indent all the file. There are different indentations along the file.

File: include/fastrtps/qos/LivelinessChangedStatus.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+#include <fastrtps/rtps/common/InstanceHandle.h>
+
+namespace eprosima {
+namespace fastrtps {
+
+//! @brief A struct storing the liveliness changed status
+struct LivelinessChangedStatus
+{
+    //! @brief Constructor
+    LivelinessChangedStatus()
```
Comment (richiware) [inline include/fastrtps/qos/LivelinessChangedStatus.h]: You can remove it and use the implicit default constructor.
Also in C++11 you can initialize basic types in their definitions.

File: include/fastrtps/qos/LivelinessLostStatus.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+#include <cstdint>
+
+namespace eprosima {
+namespace fastrtps {
+
+//! @brief A struct storing the liveliness lost status
+struct LivelinessLostStatus
+{
+    //! @brief Constructor
+    LivelinessLostStatus()
```
Comment (richiware) [inline include/fastrtps/qos/LivelinessLostStatus.h]: You can remove it.

File: include/fastrtps/publisher/PublisherListener.h (hunk 1 above)
Comment (raquelalvarezbanos) [reply include/fastrtps/publisher/PublisherListener.h]: Done

File: include/fastrtps/qos/LivelinessChangedStatus.h (hunk 2 above)
Comment (raquelalvarezbanos) [reply include/fastrtps/qos/LivelinessChangedStatus.h]: :+1:

File: include/fastrtps/qos/LivelinessLostStatus.h (hunk 3 above)
Comment (raquelalvarezbanos) [reply include/fastrtps/qos/LivelinessLostStatus.h]: :+1:

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1502/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1467/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1478/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1508/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1473/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1484/)

**Comment by richiware** [comment]
Can't set status; build failed.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1522/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1485/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1496/)

**Pull request merged**